### PR TITLE
create dateParse helper function in order to avoid invalid dates

### PR DESCRIPTION
**Pull request by aurel-l**
## Purpose
Fix issue with some date not being parsed correctly and throwing error on some pages (/proteomes/UP000149066) in some browsers (Firefox and Safari)

## Approach
Create a wrapper `parseDate` around date parsing and handles some edge cases that we know of.
Makes sure that even if it cannot parse the date, it doesn't return an invalid date that would cause an issue further down the line.
Replace all bare uses of `new Date()` with the new wrapper.

## Testing
Added tests for wrapper (note that it's running on V8, so would have the same behaviour as Firefox or Safari).
Made sure the rest of tests were running, and that the problematic page was not crashing in Firefox

## Checklist
- [x] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

### CODE REVIEW

File: src/shared/utils/parseDate.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  }
+  // easy case
+  let output = new Date(input);
+  if (!Number.isNaN(output.getTime())) {
+    return output;
+  }
+  /* edge cases */
+  // This one is specific to Firefox and Safari, as of April 2021.
+  // This will not be able to be covered by tests as node is using the V8 engine
+  if (typeof input === 'string' && edgeCase1.test(input)) {
```
Comment (dlrice) [inline src/shared/utils/parseDate.ts:38]: Could extract into a separate function (eg `parseDifficultDate`) to run unit tests in V8

File: src/shared/utils/parseDate.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    const [month, year] = input.split('-');
+    const monthIndex = months.indexOf(month.toLowerCase());
+    output = new Date(+year, monthIndex);
+    if (monthIndex >= 0 && !Number.isNaN(output.getTime())) {
+      return output;
+    }
+  }
+  // otherwise just return undefined
+  return undefined;
```
Comment (dlrice) [inline src/shared/utils/parseDate.ts:47]: How about just returning the provided string as a fallback? I realise this will be a bit tricky as we call `toISOString` on the output of this fn in most places.  One idea then is creating a fn `parseAndFormatDate` which calls `parseDate` and if the return of this is `undefined` returns the provided string and otherwise immediately calls `toISOString`?

Comment (dlrice) [reply src/shared/utils/parseDate.ts:47]: A console.error that things didn't work out could be useful to find these unsuccessful instances in the future

File: src/shared/utils/parseDate.ts (hunk 1 above)
Comment (aurel-l) [reply src/shared/utils/parseDate.ts]: Good idea, I did that, and that even allowed me to catch an issue in my regular expression 👍🏽

File: src/shared/utils/parseDate.ts (hunk 2 above)
Comment (aurel-l) [reply src/shared/utils/parseDate.ts:59]: I added the console.error, it would be good to other edge cases eventually too.

Comment (aurel-l) [reply src/shared/utils/parseDate.ts:59]: About returning the provided string, I don't think it is necessary. In the places where we use a string as source of information for a data, we use display the unmodified string to the user. The date is only parsed as an extra information passed to the `<time>` tag shown to the user's assistive technology, or a bot browsing the page. That's also the reason why I thought it was OK to miss parsing some cases, because worst case scenario we still do display the original string to the user and the user "parses" the data in their mind, even if the browser doesn't recognise it as a date

**Review by dlrice — APPROVED** [review]
